### DISCUSSION TITLE
Beatmap offset not resetting on restarting client

### DISCUSSION THREAD

**Discussion by NatureHog** [ppy/osu#22509] [Q&A] [answered]
I sometimes use the "use offset from previous play" button for specific beatmaps, so the next time I would play said beatmap, even after the client was closed, the same offset is used, which is very confusing for me...

Is there an option to "reset" offsets for each individual beatmap whenever I exit the client? Or am I doing something wrong?

I'm using version 2023.123.0

Not sure if really needed but here are the (zipped) logs, before and after exiting and re-launching the game:
[logs-before-closing.zip](https://github.com/ppy/osu/files/10578413/logs-before-closing.zip)
[logs-after-closing.zip](https://github.com/ppy/osu/files/10578416/logs-after-closing.zip)

Beatmap I was testing with, just in case:
https://osu.ppy.sh/beatmapsets/1094838#osu/2295079

**Comment by mk56-spn** [comment, marked as answer]
Why would you want to reset the offsets that are fixing your deviation on exit? 

Im somewhat perplexed by this use case, care to elaborate?

**Reply by NatureHog** [reply]
I usually don't play the same beatmap over and over again, I normally just take some random beatmap until I'm satsfied with the score, then move on to the next one.
That means each "session" consists of (mostly) different beatmaps, which means sometimes I only play the same beatmap every few sessions or so.

When this happens, I usually already forgot about the offset, and keep playing with the offset, which is very confusing.

Or maybe I'm just doing this wrong...

**Reply by mk56-spn** [reply]
But the auto offset is per map?

**Reply by NatureHog** [reply]
Yes, so each time I play the same map, the offset remains the same, which usually was a few days ago...

Edit to clarify: I (personally) don't find it beneficial to have the same offset for the same beatmap, which might've been 5~6 days ago...

**Reply by mk56-spn** [reply]
why shouldn't it remain the same? unless the hardware/ audio settings you are using have been altered said offset should still be correct, whether that be over an hour, a day or a year

**Reply by NatureHog** [reply]
Each play I do is so different (especially UR), that it _feels_ wrong, at least. Or I could be doing this all wrong, and I'm not quite using it right...

**Reply by bdach** [reply]
The offset was intentionally designed to be persistent for the given beatmap. This is especially useful with mistimed maps where you don't want to be setting the same offset every play.

**Reply by monochrome22** [reply]
But resetting the offset makes it feel perfect? Even if you could reset all the offsets, wouldn't you still constantly be setting a new offset, meaning resetting them didn't really accomplish anything? 

Having a button in settings to "Reset all beatmap offsets" does sound like something that you'd expect to exist, but I can't think of any practical uses for it. 

It actually doesn't sound completely unreasonable that someone would require an offset when learning a new map and then no longer needing it once they figure out the proper timings, but it really doesn't sound like a button to reset all offsets would be necessary, even for this rare case. Just making sure you know; you can click the little green dot next to beatmap offset to reset the offset. 

Also, I'm 95% sure that your problem is caused by the fact that you are still learning to sense the timings and follow the rhythm, I hope this doesn't sound rude.

**Reply by NatureHog** [reply]
Thanks everyone.

As it turns out, I misunderstood the core idea behind offsets.
